tsweb/tsweb.go

```diff
@@ -466,6 +466,12 @@ func writePromExpVar(w io.Writer, prefix string, kv expvar.KeyValue) {
 		}
 		fmt.Fprintf(w, "# TYPE %s %s\n%s %v\n", name, typ, name, v.Value())
 		return
+	case *expvar.Float:
+		if typ == "" {
+			typ = "gauge"
+		}
+		fmt.Fprintf(w, "# TYPE %s %s\n%s %g\n", name, typ, name, v.Value())
```

Comment:
why the %g in particular? and all the testing for `e` below?

https://github.com/Showmax/prometheus-docs/blob/master/content/docs/instrumenting/exposition_formats.md says anything that strconv.ParseFloat can parse.

which means anything works, including the good ol' %v default: https://go.dev/play/p/CwQ7xQK-oay

Reply:
That is fair. I had been comparing to derp-bwmon's current `prometheus/client_golang` metrics to keep the output roughly the same, but it doesn't need to exactly match so long as Prometheus can scrape it.

Reply:
We use `prometheus/client_golang` anywhere!? :)